File: dart/s5_problem_solving_approaches/search_divide_and_conquer.dart
int search (List<int> list, int value) {
  int min = 0;
  int max = list.length - 1;

  while (min <= max) {
    int middle = ((min + max) / 2).floor();

    if (list[middle] < value) {
      min = middle + 1;
    } else if (list[middle] > value) {
      max = middle - 1;
    } else {
      return middle;
    }
  }

  return -1;
}

void main() {
  print(search([1,3,6,12,55,78],56));
}

File: js/s5-problem-solving-approaches/search-divide-and-conquer.js
function search(array, value) {
    let min = 0;
    let max = array.length - 1;

    while (min <= max) {
        let middle = Math.floor((min + max) / 2);

        if (array[middle] < value) {
            min = middle + 1;
        } else if (array[middle] > value) {
            max = middle - 1;
        } else {
            return middle;
        }
    }
    return -1;
}

console.log(search([1,3,6,12,55,78],12))